lib/internal/http2/compat.js

```diff
@@ -668,21 +666,19 @@ class Http2ServerResponse extends Stream {
     if (chunk !== null && chunk !== undefined)
       this.write(chunk, encoding);
 
-    const isFinished = this.finished;
-    state.headRequest = stream.headRequest;
     state.ending = true;
 
     if (typeof cb === 'function') {
-      if (isFinished)
+      // TODO(ronag): ERR_STREAM_ALREADY_FINISHED
+      if (stream.writableFinished)
         this.once('finish', cb);
       else
         stream.once('finish', cb);
     }
 
     if (!stream.headersSent)
       this.writeHead(this[kState].statusCode);
-
-    if (isFinished)
+    else if (stream.writableEnded)
       onStreamCloseResponse.call(stream);
```

Comment:
@apapirovski I don't quite understand this branch? How could the response be finished/ended before calling `end()`? My gut feeling is that this is not required.